drivers/spi/spi_andes_atcspi200.c

```diff
@@ -0,0 +1,629 @@
+	struct spi_atcspi200_data * const data = DEV_DATA(dev);
+	struct spi_context *ctx = &data->ctx;
+	uint32_t rx_data, cur_tx_fifo_num, cur_rx_fifo_num;
+	uint32_t i, dfs, intr_status, spi_status;
+	uint32_t tx_num = 0, tx_data = 0;
+
+	intr_status = sys_read32(SPI_INTST(dev));
+	dfs = SPI_WORD_SIZE_GET(ctx->config->operation) >> 3;
+
+	if ((intr_status & REG_INTST_TX_FIFO_INT_MSK) &&
+	    !(intr_status & REG_INTST_END_INT_MSK)) {
+
+		spi_status = sys_read32(SPI_STAT(dev));
+		cur_tx_fifo_num = GET_TX_NUM(dev);
+
+		tx_num = data->tx_fifo_size - cur_tx_fifo_num;
+
+		for (i = tx_num; i > 0; i--) {
+
+			if (data->tx_cnt >= data->chunk_len) {
+				/* Have already sent a chunk of data, so stop
+				 * sending data!
+				 */
+				CLR_MASK(SPI_INTEN(dev), REG_IEN_TX_FIFO_MSK);
+				break;
+			}
+
+			if (spi_context_tx_buf_on(ctx)) {
+
+				switch (dfs) {
+				case 1:
+					tx_data = *ctx->tx_buf;
+					break;
+				case 2:
+					tx_data = *(uint16_t *)ctx->tx_buf;
+					break;
+				}
+
+			} else if (spi_context_tx_on(ctx)) {
+				tx_data = 0;
+			} else {
+				CLR_MASK(SPI_INTEN(dev), REG_IEN_TX_FIFO_MSK);
+				break;
+			}
+
+			sys_write32(tx_data, SPI_DATA(dev));
+
+			spi_context_update_tx(ctx, dfs, 1);
+
+			data->tx_cnt++;
+		}
+		sys_write32(REG_INTST_TX_FIFO_INT_MSK, SPI_INTST(dev));
+
+	}
+
+	if (intr_status & REG_INTST_RX_FIFO_INT_MSK) {
+		cur_rx_fifo_num = GET_RX_NUM(dev);
+
+		for (i = cur_rx_fifo_num; i > 0; i--) {
+
+			rx_data = sys_read32(SPI_DATA(dev));
+
+			if (spi_context_rx_buf_on(ctx)) {
+
+				switch (dfs) {
+				case 1:
+					*ctx->rx_buf = rx_data;
+					break;
+				case 2:
+					*(uint16_t *)ctx->rx_buf = rx_data;
+					break;
+				}
+
+			} else if (!spi_context_rx_on(ctx)) {
+				CLR_MASK(SPI_INTEN(dev), REG_IEN_RX_FIFO_MSK);
+			}
+
+			spi_context_update_rx(ctx, dfs, 1);
+		}
+		sys_write32(REG_INTST_RX_FIFO_INT_MSK, SPI_INTST(dev));
+	}
+
+	if (intr_status & REG_INTST_END_INT_MSK) {
+
+		/* Clear end interrupt */
+		sys_write32(REG_INTST_END_INT_MSK, SPI_INTST(dev));
+
+		/* Disable all SPI interrupts */
+		sys_write32(0, SPI_INTEN(dev));
+
+		transfer_next_chunk(dev);
+	}
+}
+
+#define SPI_DMA_CHANNEL(id, dir, DIR, src, dest)
+#define SPI_IF_NO_CMD(num) .is_cmdaddr_mode = 0,
+#define SPI_BUSY_INIT .busy = false,
+
+#if (CONFIG_XIP)
+#define SPI_ROM_CFG_XIP(node_id) DT_SAME_NODE(node_id, DT_BUS(DT_CHOSEN(zephyr_flash)))
+#else
+#define SPI_ROM_CFG_XIP(node_id) false
+#endif
+
+#define SPI_INIT(n)							\
+	static struct spi_atcspi200_data spi_atcspi200_dev_data_##n = {	\
+		SPI_CONTEXT_INIT_LOCK(spi_atcspi200_dev_data_##n, ctx),	\
+		SPI_CONTEXT_INIT_SYNC(spi_atcspi200_dev_data_##n, ctx),	\
+		SPI_IF_NO_CMD(n)					\
+		SPI_BUSY_INIT						\
+		SPI_DMA_CHANNEL(n, rx, RX, PERIPHERAL, MEMORY)		\
+		SPI_DMA_CHANNEL(n, tx, TX, MEMORY, PERIPHERAL)		\
+	};								\
+	static void spi_atcspi200_cfg_##n(void);			\
+	static struct spi_atcspi200_cfg spi_atcspi200_dev_cfg_##n = {	\
```

Comment:
```suggestion
	static const struct spi_atcspi200_cfg spi_atcspi200_dev_cfg_##n = {	\
```

Reply:
Thanks for reviewing!

Reply:
It seems that you have forgotten about that.

Reply:
It my mistake, thanks for your remind.